h2o-algos/src/main/java/hex/svd/SVD.java

```diff
@@ -98,7 +100,7 @@ protected void checkMemoryFootPrint() {
       error("_train", msg);
     }
 
-    if (mem_usage > max_mem) {  // choose the most memory efficient one
+    if (mem_usage > mem_usage_w) {  // choose the most memory efficient one
```

Comment:
I am struggling with this one a bit.  In theory, I only use SVD when it is being called from PCA.  However, we do have documentations that show SVD as a class of its own.  If people call SVD on its own, it will need to be able to choose the more memory efficient mode as well.  If we can guarantee that people only calls SVD from PCA, then it is duplicated.